### DISCUSSION TITLE
Limit results of a Contacts data query to the ones matching the addresses of the current ticket +  Merge Contacts

### DISCUSSION THREAD

**Discussion by j-at-se** [cerb/cerb-release#82] [Q&A]
Hi!

I'm working on a new widget that will display the ticket contacts (as opposed to displaying the ticket participants).

So far I'm happy with my progress but I'm hitting a wall: the widget displays every single contact we have on record.

Here's the widget:
```
{
    "widget": {
        "uid": "profile_widget_154",
        "_context": "cerberusweb.contexts.profile.widget",
        "name": "Contacts (In Development)",
        "extension_id": "cerb.profile.tab.widget.sheet",
        "pos": 4,
        "width_units": 4,
        "zone": "sidebar",
        "extension_params": {
            "data_query": "type:worklist.records\r\nof:contact\r\nexpand:[]\r\nquery:(\r\n  email:[{{ticket_participants}}]\r\n  sort:email\r\n  limit: 5\r\n)\r\nformat:dictionaries",
            "cache_secs": "0",
            "placeholder_simulator_kata": "record_id: 1",
            "sheet_kata": "layout:\r\n  style: table\r\n  headings@bool: no\r\n  paging@bool: yes\r\n  filtering@bool: yes\r\n\r\ncolumns:\r\n  selection/id:\r\n  card/email:\r\n    label: Contact\r\n    params:\r\n      image@bool: yes\r\n      bold@bool: yes\r\n      context_key: _context\r\n      label_key: _label\r\n      id_key: id",
            "toolbar_kata": "interaction/copy:\r\n  label: Copy emails\r\n  uri: cerb:automation:cerb.interaction.echo\r\n  icon: copy\r\n  inputs:\r\n    outputs:\r\n      clipboard: {{record_participants|column('email')|join(', ')}}\r\n      alert: Participants' emails copied to clipboard.\r\n      \r\ninteraction/merge:\r\n  label: Merge {{row_selections|length}} contacts\r\n  uri: cerb:automation:cerb.IsThereNoMergeAutomation\r\n  inputs:\r\n    ticket@key: record_id \r\n    participants@key: row_selections \r\n  icon: circle-remove\r\n  hidden@bool: \r\n    {% if row_selections is empty %}yes{% endif %}\r\n  after:\r\n    refresh_widgets@list:\r\n      Participants"
        }
    }
}
```
![image](https://github.com/user-attachments/assets/25136863-3585-4699-8691-407b3ccf478c)

Also, I can't seem to find an automation for a "merge contacts" interaction. I know it's possible to do it manually so I'm wondering if maybe it isn't supported through automation?

Thank you in advance.

**Comment by jstanden** [comment]
Hi @j-at-se!

For your data query, you'd want to use the `record_` placeholders. There isn't a `ticket_` placeholder in the scope on profiles.

```yaml
type:worklist.records
of:contact
expand:[]
query:(
  email:[{{record_participants|column('email')|join(',')}}]
  sort:email
  limit: 5
)
format:dictionaries
```

As for merging records from automations, we've considered adding an [api.command](https://cerb.ai/docs/automations/commands/api.command/) for doing it procedurally.

I also think it'd be useful to be able to trigger the merge popup from interactions. That could be similar to [await:record:](https://cerb.ai/docs/automations/triggers/interaction.worker/#awaitrecord)

**Reply by j-at-se** [reply]
Yup! This worked as expected. I knew it had to be something simple.

As for merging, I understand it's not currently possible from automations. I would say the await:record: option makes the most sense to me, but I can't complain about having multiple ways to merge records.

I'd be interested in beta testing record merge when it's added to Cerb.

Now, when I have more than one participant in the ticket, the widget doesn't display a single contact. I assume this is due to the emails (e.g. "johnsmith@email.com,jsmith@email.com") being taken as a single value and therefore not finding a match. How do you resolve this?
